File: hisi-dev-tool-frontend/src/api/projectNameGroup.ts
import request from '@/utils/request'

export interface ProjectNameGroup {
  id?: number
  groupName: string
  groupPattern: string
  projectNames: string[]
  description?: string
  createdAt?: string
  updatedAt?: string
}

export const projectNameGroupApi = {
  // 获取所有分组
  getGroups(): Promise<ProjectNameGroup[]> {
    return request.get('/project-name-group') as Promise<ProjectNameGroup[]>
  },

  // 获取单个分组
  getGroup(groupName: string): Promise<ProjectNameGroup> {
    return request.get(`/project-name-group/${groupName}`) as Promise<ProjectNameGroup>
  },

  // 创建或更新分组
  saveGroup(group: ProjectNameGroup): Promise<ProjectNameGroup> {
    return request.post('/project-name-group', group) as Promise<ProjectNameGroup>
  },

  // 删除分组
  deleteGroup(groupName: string): Promise<void> {
    return request.delete(`/project-name-group/${groupName}`)
  }
}

File: hisi-dev-tool/src/main/java/com/huawei/hisi/project/namegroup/model/ProjectNameGroup.java
package com.huawei.hisi.project.namegroup.model;

import java.time.LocalDateTime;
import java.util.List;

import lombok.Data;

/**
 * 项目名称分组实体
 * Task 74: 按项目名称前缀/模式分组
 */
@Data
public class ProjectNameGroup {

    private Long id;
    private String groupName;      // 分组名称，如 "HiSi DevTool 系列"
    private String groupPattern;   // 分组模式，如 "hisi-*"
    private List<String> projectNames;  // 分组下的项目名称列表
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}